.github/workflows/create-release.yaml

```diff
@@ -1,153 +1,345 @@
+          output = subprocess.check_output(command, stderr=subprocess.STDOUT)
+          if output.decode() != "":
+            print(f"::warning::Tag '{tag_name}' already exists, will skip creation")
+            # Output for downstream steps to check
+            with open(os.environ['GITHUB_OUTPUT'], 'a') as f:
+              f.write(f"tag_exists=true\n")
+          else:
+            print(f"Tag '{tag_name}' does not exist, will create it")
+            with open(os.environ['GITHUB_OUTPUT'], 'a') as f:
+              f.write(f"tag_exists=false\n")
+
+        shell: python
+        continue-on-error: false
+
+      - name: Log tag exists - skip creation
+        if: ${{ steps.check_tag.outputs.tag_exists == 'true' }}
+        run: |
+          echo "::notice::Tag ${{ github.event.inputs.tag_name }} already exists, no further action will be taken"
+
+      - name: Prepare params.env update
+        if: ${{ steps.check_tag.outputs.tag_exists != 'true' }}
+        run: |
+          echo "::notice::Tag ${{ github.event.inputs.tag_name }} will be created"
+          echo "::notice::Updating config/base/params.env with new tag"
+
+          # Update params.env with release tag
+          sed -i "s|odh-model-controller:fast|odh-model-controller:${{ github.event.inputs.tag_name }}|g" config/base/params.env
+
+          # Check and log if params.env will be updated
+          if git diff --quiet config/base/params.env; then
+            echo "::notice::No changes needed in config/base/params.env"
+          else
+            echo "::notice::config/base/params.env will be updated on the new tag"
+            git diff config/base/params.env
+          fi
+
+      - name: Create and push tag
+        if: ${{ github.event.inputs.dry_run != 'true' && steps.check_tag.outputs.tag_exists != 'true' }}
+        run: |
+          git config --global user.email "github-actions@github.com"
+          git config --global user.name "GitHub Actions"
+          git add config/base/params.env
+
+          # Commit changes if there are any
+          if [[ -n "$(git status --porcelain)" ]]; then
+            git commit -m "Update image refs for ODH release ${{ github.event.inputs.tag_name }}"
+          fi
+
+          # Create and push tag
+          git tag -a ${{ github.event.inputs.tag_name }} -m "Prepare for ODH release ${{ github.event.inputs.tag_name }}"
+          git push origin ${{ github.event.inputs.tag_name }}
+
+  changelog:
+    name: Changelog
+    needs: fetch-tag
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+
+      - name: Check if release exists
+        id: check_release
+        run: |
+          if gh release view "${{ github.event.inputs.tag_name }}" >/dev/null 2>&1; then
+            echo "release_exists=true" >> $GITHUB_OUTPUT
+            echo "::warning::Release already exists for tag ${{ github.event.inputs.tag_name }}, skipping creation"
```

Comment:
shouldn't we abort if the release already exists?

Also I think we should clean anything that was created so far (if any)

Reply:
I am more concerned about possible garbage left behind by failures.